Sources/Core/Yoakke.Reporting/Present/TextDiagnosticsPresenter.cs

```diff
@@ -63,6 +63,28 @@ public TextDiagnosticsPresenter(TextWriter writer)
             this.buffer = new ColoredBuffer();
         }
 
+        private static IEnumerable<LinePrimitive> TrimEmptySourceLinesAtEdges(List<LinePrimitive> linePrimitives)
+        {
+            static bool IsEmptySourceLine(LinePrimitive line)
+            {
+                if (line is SourceLine sourceLine)
+                {
+                    var contents = sourceLine.Source!.GetLine(sourceLine.Line);
+                    return string.IsNullOrWhiteSpace(contents);
+                }
+                else
+                {
+                    return false;
+                }
```

Comment:
I think guard conditions simplify the code a bit here, something like this:
```cs
if (line is not SourceLine sourceLine) return false;
```

Reply:
No longer needed as of last commit, as that code has been removed.